### PR TITLE
[REF] project_timesheet_holidays: hooks

### PR DESCRIPTION
**Pull request by alexey-pelykh**
Description of the issue/feature this PR addresses:
Addresses #30854 and regular code maintenance.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/project_timesheet_holidays/tests/test_timesheet_holidays.py [hunk 1]
```diff
@@ -85,7 +83,7 @@ def test_validate_with_timesheet(self):
             'number_of_days': number_of_days,
         })
         holiday.sudo().action_validate()
-        self.assertEquals(len(holiday.timesheet_ids), number_of_days, 'Number of generated timesheets should be the same as the leave duration (1 per day between %s and %s)' % (fields.Datetime.to_string(self.leave_start_datetime), fields.Datetime.to_string(self.leave_end_datetime)))
+        self.assertEquals(len(holiday.timesheet_ids), number_of_days, 'Number of generated timesheets should be the same as the leave duration (1 per day between %s and %s)' % (self.leave_start_datetime, self.leave_end_datetime))
```
Comment (pedrobaeza) [inline addons/project_timesheet_holidays/tests/test_timesheet_holidays.py]: Other thing that should be in an isolated PR

Comment (alexey-pelykh) [reply addons/project_timesheet_holidays/tests/test_timesheet_holidays.py]: 😩

Comment (pedrobaeza) [reply addons/project_timesheet_holidays/tests/test_timesheet_holidays.py]: The concept is simple: one change per PR :wink:

Comment (alexey-pelykh) [reply addons/project_timesheet_holidays/tests/test_timesheet_holidays.py]: #30872

File: addons/project_timesheet_holidays/models/hr_holidays.py [hunk 2]
```diff
@@ -65,32 +95,14 @@ def _validate_leave_request(self):
...
-                })
-
-        return super(Holidays, self)._validate_leave_request()
+        return super()._validate_leave_request()
 
     @api.multi
     def action_refuse(self):
         """ Remove the timesheets linked to the refused holidays """
-        result = super(Holidays, self).action_refuse()
+        result = super().action_refuse()
```
Comment (mreficent) [inline addons/project_timesheet_holidays/models/hr_holidays.py]: I think this kind of change shouldn't be in the PR... Same for the other super above.

Comment (alexey-pelykh) [reply addons/project_timesheet_holidays/models/hr_holidays.py]: On other hand, why to keep useless code in codebase? Or wait for Odoo employees to do that?

Comment (mreficent) [reply addons/project_timesheet_holidays/models/hr_holidays.py]: Yeah, they do that on future refactorings and cleaning ups...

**Pull request closed without merging**